src/main/java/strategy/handlers/BalanceHandler.java

```diff
@@ -0,0 +1,12 @@
+package strategy.handlers;
+
+import db.Storage;
+import model.FruitTransaction;
+import strategy.OperationHandler;
+
+public class BalanceHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        Storage.fruitStorage.put(transaction.getFruit(), transaction.getQuantity());

```

Comment:
It's important to validate the `quantity` in the `transaction` before updating the storage. Ensure that the quantity is not negative to avoid incorrect states in the storage.